### PR TITLE
feat: add Funnel and FunnelChart Component

### PR DESCRIPTION
**Pull request by dhrjarun**
issue #407 

Review it and let me know What to do next

### CODE REVIEW

**Comment by archiewood** [comment]
Hey @dhrjarun,

Thanks for the PR!

I havent tested this yet, but a couple of implementation thoughts:
- The `<Chart/>` component is designed to be a base for visualizations that have x-y axes. this is probably why you've ended up needing to define more props inside this component to get it to work (eg valueCols). This isn't particularly suitable for a chart with no clear axes like a funnel. 
   - It would probably be easierand cleaner to have a <FunnelChart/> call <ECharts/> directly
   - [This might be helpful](https://docs.evidence.dev/features/charts/custom-charts#funnel-chart)
- We require changesets in PRs as that make sure all dependencies are updated correctly for npm.

**Comment by dhrjarun** [comment]
Hey @archiewood  I have made the required changes and added changeset

**Comment by archiewood** [comment]
Hey @dhrjarun can you add a page to the docs too for this component?

This might be a good example to follow: https://docs.evidence.dev/features/charts/area-chart

A couple of screenshots, and a page with a reference of all the props that the user can customize.

File: sites/docs/docs/features/charts/funnel-chart.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+hide_title: true
+hide_table_of_contents: false
+---
+
+<h1 class="community-header"><span class="gradient">&lt;Funnel Chart/></span></h1>
+
+![funnel](/img/exg-funnel-nt.svg)
+
+```markdown
+<AreaChart
```
Comment (archiewood) [inline sites/docs/docs/features/charts/funnel-chart.md:13]: Should this say "FunnelChart"?

File: sites/docs/docs/features/charts/funnel-chart.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+```
+
+## Examples
+
+### Ascending
+
+![funnel](/img/exg-funnel-ascending-nt.svg)
+
+```markdown
+<AreaChart
```
Comment (archiewood) [inline sites/docs/docs/features/charts/funnel-chart.md:27]: Also this - should this say "FunnelChart"?

File: sites/docs/docs/features/charts/funnel-chart.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    funnelSort=ascending
+/>
+```
+
+### Alignment
+
+![funnel](/img/exg-funnel-left-aligned-nt.svg)
+
+```markdown
+<AreaChart
```
Comment (archiewood) [inline sites/docs/docs/features/charts/funnel-chart.md:40]: and this

**Review by archiewood — COMMENTED** [review]
A couple of typos in the docs I think?

**Pull request merged**